Sources/PackageGraph/PackageGraph.swift

```diff
@@ -80,26 +82,29 @@ public struct PackageGraph {
             }
         }))
 
-        // Compute the input targets.
-        let inputTargets = inputPackages.flatMap({ $0.targets }).map(ResolvedTarget.Dependency.target)
-        // Find all the dependencies of the root targets.
-        let dependencies = try! topologicalSort(inputTargets, successors: { $0.dependencies })
-
-        // Separate out the products and targets but maintain their topological order.
-        var reachableTargets: Set<ResolvedTarget> = []
-        var reachableProducts = Set(inputPackages.flatMap({ $0.products }))
-
-        for dependency in dependencies {
-            switch dependency {
-            case .target(let target):
-                reachableTargets.insert(target)
-            case .product(let product):
-                reachableProducts.insert(product)
-            }
-        }
+        // Compute the reachable targets and products.
+        let inputTargets = inputPackages.lazy.flatMap({ $0.targets })
```

Comment:
Is lazy actually doing anything in this situation?

Reply:
You're right. I've moved lazy down to where it actually makes sense.